tests/APNSTest.php

```diff
@@ -40,13 +40,17 @@ function new_alert(): APNSAlert {
 		return new APNSAlert( $this->random_string(), $this->random_string() );
 	}
 
-	function new_metadata( ?string $topic = null ): APNSRequestMetadata {
+	function new_metadata( ?string $topic = null, ?string $uuid = null ): APNSRequestMetadata {
 
 		if ( is_null( $topic ) ) {
 			$topic = $this->random_string();
 		}
 
-		return new APNSRequestMetadata( $topic );
+		if ( is_null( $uuid ) ) {
+			$uuid = $this->random_uuid();
+		}
```

Comment:
PHP question: I'm guessing we're doing the assignment here instead of doing it in the default value setup above because it call `$this` which can't be used when setting the default value in the function definition?

What I mean is

```php
function foo( string $uuid = $this->random_uuid() ) {
```

Reply:
That's exactly right, yep. No expressions are allowed there – only values